lib/spack/llnl/util/symlink.py

```diff
@@ -42,21 +42,23 @@ def symlink(real_path, link_path):
 
 
 def islink(path):
-    return os.path.islink(path) or _win32_is_junction(path)
+    return Path(path).is_link() or _win32_is_junction(path)
 
 
 # '_win32' functions based on
 # https://github.com/Erotemic/ubelt/blob/master/ubelt/util_links.py
 def _win32_junction(path, link):
+    path = Path(path)
+    link = Path(link)
     # junctions require absolute paths
-    if not os.path.isabs(link):
-        link = os.path.abspath(link)
+    if not link.is_absolute():
+        link = link.absolute()
```

Comment:
`Path.absolute()` already checks `.is_absolute()`, so we can avoid doing that again here.

```suggestion
    link = link.absolute()
```

Reply:
Cool! Thanks for digging into that to verify the behavior.